include/common/bitdepth.rs

```diff
@@ -96,6 +96,17 @@ pub trait BitDepth: Clone + Copy {
 
     fn new(bitdepth_max: Self::BitDepthMax) -> Self;
 
+    /// While [`Self::new`] is the implementation specific way to construct a [`BitDepth`],
+    /// [`Self::from_c`] is a uniform way to construct a [`BitDepth`]
+    /// from its C representation, which is a [`c_int`].
+    ///
+    /// Since we're deduplicating [`BitDepth`]-dependent `fn` ptr types through type erasure,
+    /// and always passing the `bitdepth_max` last argument
+    /// even for `8bpc` `fn`s where it's superfluous (it's constant),
+    /// we need to convert from that `bitdepth_max: c_int` arg back to a [`BitDepth`],
+    /// which is what this `fn` is for.
```

Comment:
> which is what this `fn` is for.

redundant. 

General comment: you should wrap long comments consistently to the text looks justified around col 80.

```
    /// While [`Self::new`] is the implementation specific way to construct a [`BitDepth`],
    /// [`Self::from_c`] is a uniform way to construct a [`BitDepth`] from its C representation
    /// representation, which is a [`c_int`].
    ///
    /// Since we're deduplicating [`BitDepth`]-dependent `fn` ptr types through type erasure,
    /// and always passing the `bitdepth_max` last argument even for `8bpc` `fn`s where
    /// it's superfluous (it's constant), we need to convert from that `bitdepth_max: c_int` 
    /// arg back to a [`BitDepth`],
```

> Since we're deduplicating

Comments should just document what the function does, not talk about the development process.

Reply:
> > which is what this `fn` is for.
> 
> redundant.

👍

> General comment: you should wrap long comments consistently to the text looks justified around col 80.

Oh, I always try to wrap them at semantic breaks.  That way you get much cleaner diffs when you edit the comments.  I can wrap them at 80 columns, but I'd prefer to wrap at semantic breaks rather than hard wrapping at word boundaries.

> Comments should just document what the function does, not talk about the development process.

I thought it's helpful to explain why we convert certain C constructs to different Rust ones, like this interface between the Rust API and the C way of doing it.

Reply:
> ```
>     /// While [`Self::new`] is the implementation specific way to construct a [`BitDepth`],
>     /// [`Self::from_c`] is a uniform way to construct a [`BitDepth`] from its C representation
>     /// representation, which is a [`c_int`].
>     ///
>     /// Since we're deduplicating [`BitDepth`]-dependent `fn` ptr types through type erasure,
>     /// and always passing the `bitdepth_max` last argument even for `8bpc` `fn`s where
>     /// it's superfluous (it's constant), we need to convert from that `bitdepth_max: c_int` 
>     /// arg back to a [`BitDepth`],
> ```

I wasn't sure if this was a suggestion since some of the lines are longer than 80, but I changed the comment to wrap at column 80 now.  And I tried to change the language so it's not talking about the development process, but more so how things are implemented.  Does that look better now?